card/mod/history/set/all/history.rb

```diff
@@ -83,7 +83,7 @@ def finalize_action?
 event :remove_empty_act,
       :integrate_with_delay_final,
       when: proc { |c|  c.act_card? } do
-  if @current_act&.actions&.reload.empty?
+  if @current_act&.actions&.reload&.empty?
```

Comment:
generally speaking, I think these kinds of things should be in a :when test.

(I know you didn't add it here)

The question is whether something an `after: :remove_empty_act` event should fire, even if no empty act was removed.  If this is in `when:`, it doesn't fire.

In fact, we might even start a pattern where if I make `def remove_empty_act?`, then this is automatically treated as the test?  hmm...

Reply:
Sounded like a good idea but becomes worse the more you think about it. The only win is that we save a few key strokes but if the check method isn't close to the event the logic is obscured and it will become a nightmare of debugging if you create a check method that accidentally matches an existing event or vice versa.

Reply:
Yeah, I agree.  maybe just a convention (specified explicitly)